File: LLM-Translations/GPT-4/codenet/Python/Java/s259116349.java
import java.util.*;

public class s259116349 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();

        int[] primes = {2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47};
        int[][] divs = new int[100][primes.length];
        
        for (int i = 1; i <= 100; i++) {
            for (int j = 0; j < primes.length; j++) {
                int p = primes[j];
                int tmp = 0;
                while (i % p == 0) {
                    i /= p;
                    tmp++;
                }
                divs[i - 1][j] = tmp;
            }
        }
        
        int[] divsum = new int[primes.length];
        for (int i = 1; i <= N; i++) {
            for (int j = 0; j < primes.length; j++) {
                divsum[j] += divs[i - 1][j];
            }
        }
        
        int ans = 0;

        // >=74
        ans += num(75, divsum);

        // >=14 & 13~4
        ans += num(15, divsum) * (num(5, divsum) - 1);

        // >=24 & 23~2
        ans += num(25, divsum) * (num(3, divsum) - 1);

        // >=4,>=4 & >=2
        ans += num(5, divsum) * (num(5, divsum) - 1) * (num(3, divsum) - 2) / 2;
        
        System.out.println(ans);
    }

    public static int num(int n, int[] divsum) {
        int count = 0;
        for (int j : divsum) {
            if (j >= n - 1) {
                count++;
            }
        }
        return count;
    }
}
// 

File: LLM-Translations/GPT-4/codenet/Python/C/s259116349.c
#include <stdio.h>

int main() {
    int N;
    scanf("%d", &N);

    int primes[] = {2,3,5,7,11,13,17,19,23,29,31,37,41,43,47};
    int divs[100][15];
    for(int i = 1; i <= 100; i++) {
        for(int j = 0; j < 15; j++) {
            int p = primes[j];
            divs[i - 1][j] = 0;
            while(i % p == 0) {
                i /= p;
                divs[i - 1][j]++;
            }
        }
    }

    int divsum[15] = {0};
    for(int i = 1; i <= N; i++) {
        for(int j = 0; j < 15; j++) {
            divsum[j] += divs[i - 1][j];
        }
    }

    int num(int n) {
        int count = 0;
        for(int i = 0; i < 15; i++) {
            if(divsum[i] >= n - 1) {
                count++;
            }
        }
        return count;
    }

    int ans = 0;

    ans += num(75);
    ans += num(15) * (num(5) - 1);
    ans += num(25) * (num(3) - 1);
    ans += num(5) * (num(5) - 1) * (num(3) - 2) / 2;

    printf("%d\n", ans);
    return 0;
}

// 
